mflowgen/tile_array/custom-power/outputs/power-strategy-dualmesh.tcl

```diff
@@ -0,0 +1,166 @@
+#=========================================================================
+# This script implements a dual power mesh, with a fine-grain power mesh
+# on the lower metal layers and a coarse-grain power mesh on the upper
+# metal layers. Note that M2/M3 are expected to be horizontal/vertical,
+# and the lower metal layer of the coarse power mesh is expected to be
+# horizontal.
+#
+# Author : Christopher Torng
+# Date   : March 26, 2018
+
+#-------------------------------------------------------------------------
+# Shorter names from the ADK
+#-------------------------------------------------------------------------
+
+set pmesh_bot $ADK_POWER_MESH_BOT_LAYER
+set pmesh_top $ADK_POWER_MESH_TOP_LAYER
+
+#-------------------------------------------------------------------------
+# M3 power stripe settings
+#-------------------------------------------------------------------------
+# We are using addStripe to add sets of stripes (a set includes both VSS
+# and VDD).
+#
+# Set variables to space VSS and VDD straps evenly throughout the chip.
+#
+#     VSS    VDD    VSS
+#     | |    | |    | |
+#     | |    | |    | |
+#     | |    | |    | |
+#
+#     _______ <---- $M3_str_pitch
+#     ______________
+#        ____   ^
+#     ___  ^    |
+#      ^   |    +-- $M3_str_interset_pitch
+#      |   +------- $M3_str_intraset_spacing
+#      +----------- $M3_str_width
+#
+# - M3_str_intraset_spacing : Space between VSS/VDD, chosen for constant
+#                             pitch across all VSS and VDD stripes
+# - M3_str_interset_pitch   : Pitch between same-signal stripes
+#
+# - M3_str_offset           : Offset from left edge of core to center the
+#                             M3 stripe between vertical M3 routing tracks
+#
+# - M3_str_width            : M3 stripe width
+# - M3_str_pitch            : Choosing an arbitrary M3 pitch that is a
+#                             multiple of M3 signal pitch for now
+
+# Get M3 min width and signal routing pitch as defined in the LEF
+
+set M3_min_width    [dbGet [dbGetLayerByZ 3].minWidth]
+set M3_route_pitchX [dbGet [dbGetLayerByZ 3].pitchX]
+
+# Set M3 stripe variables
+
+set M3_str_width            [expr  3 * $M3_min_width]
+set M3_str_pitch            [expr 10 * $M3_route_pitchX]
+
+set M3_str_intraset_spacing [expr $M3_str_pitch - $M3_str_width]
+set M3_str_interset_pitch   [expr 2*$M3_str_pitch]
+
+set M3_str_offset           [expr $M3_str_pitch + $M3_route_pitchX/2 - $M3_str_width/2]
+
+setViaGenMode -reset
+setViaGenMode -viarule_preference default
+setViaGenMode -ignore_DRC false
+
+setAddStripeMode -reset
+setAddStripeMode -stacked_via_bottom_layer 1 \
+                 -stacked_via_top_layer    3
+
+# Ensure M3 stripes cross fully over bottom M1 stripe to prevent DRCs
+set stripeLlx [dbGet top.fPlan.coreBox_llx]
+set stripeLly [expr [dbGet top.fPlan.coreBox_lly] - [dbGet [dbGetLayerByZ 1].pitchY]]
+set stripeUrx [dbGet top.fPlan.coreBox_urx]
+set stripeUry [dbGet top.fPlan.coreBox_ury]
```

Comment:
Do you want to extend the Ury upwards like we've been doing in the others?

Reply:
It didn't cause issues, but I'll use whatever you're using in glb-top to be consistent.